techniques/systemSettings/networking/firewall/1.0/firewall.st

```diff
@@ -0,0 +1,93 @@
+      "any" usebundle => _method_reporting_context("Firewall changes", "None");
+      "any" usebundle => variable_string_from_file("firewall_technique", "firewall_conf_ref", "${reference}");
+      "any" usebundle => disable_reporting;
+      "any" usebundle => variable_string_from_command("firewall_technique", "firewall_conf", "${${firewall}.dump_cmd}");
+      "any" usebundle => enable_reporting;
+      "any" usebundle => rudder_condition_from_string_compare("rudder_firewall_technique_conf", "${firewall_technique.firewall_conf_ref}", "${firewall_technique.firewall_conf}
+");
+      "any" usebundle => enable_reporting;
+      "any" usebundle => rudder_common_report("firewall", "log_warn", "&TRACKINGKEY&", "Load configuration", "None", "Firewall configuration was modified since previous agent run"),
+                   if => canonify("variable_string_from_file_${reference}_ok.rudder_firewall_technique_conf_false");
+      "any" usebundle => condition_from_expression("rudder_firewall_conf_repaired", "true"),
+                   # needed because we cannot use the expression in a canonify directly
+                   if => canonify("file_from_string_mustache_${${firewall}.file}_repaired");
+      "any" usebundle => condition_from_expression("rudder_firewall_reload", "rudder_firewall_conf_repaired_true|rudder_firewall_technique_conf_false");
+      # Reload
+    rudder_firewall_reload_true::
+      "any" usebundle => _method_reporting_context("Load configuration", "None");
+      "any" usebundle => enable_reporting;
+      "any" usebundle => service_reload("${${firewall}.service}");
+      "any" usebundle => disable_reporting;
+      # Dump just after change (not totally atomic here)
+      "any" usebundle => variable_string_from_command("firewall_technique", "current_config_new", "${${firewall}.dump_cmd}");
+      "any" usebundle => file_content("${reference}", "${firewall_technique.current_config_new}", "true");
+      "any" usebundle => enable_reporting;
+    rudder_firewall_reload_false::
+      "any" usebundle => rudder_common_report("firewall", "result_na", "&TRACKINGKEY&", "Load configuration", "None", "Firewall configuration did not require modification");
+      # Post-hook
+      "any" usebundle => rudder_condition_from_string_compare("rudder_firewall_disable_posthook", "${posthook}", "");
+    rudder_firewall_reload_true.rudder_firewall_disable_posthook_false::
+      "any" usebundle => _method_reporting_context("Post-modification script", "None");
+      "any" usebundle => command_execution("${posthook}");
+    rudder_firewall_reload_false|rudder_firewall_disable_posthook_true::
+      "any" usebundle => rudder_common_report("firewall", "result_na", "&TRACKINGKEY&", "Post-modification script", "None", "Post-hook command was not applicable");
+    any::
+      "any" usebundle => rudder_set_force_component_none("false");
+}
+
+# Uses the systemd service, compatible with recent debian, rhel and derivatives
+bundle common rudder_nftables {
```

Comment:
this could go in another file that would be a resource, so unique

Reply:
could you also have a more specific name, as long as it's in the technique?

Reply:
What do you mean?

Reply:
rudder_firewall_nftables ?